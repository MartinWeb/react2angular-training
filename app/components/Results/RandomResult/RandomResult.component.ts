import angular from "angular";
import template from './random-result-template.html';

export const RandomResultComponent: angular.IComponentOptions = {
    template,
    bindings: {
        joke: '=',
        onReload: '<'
    }
};