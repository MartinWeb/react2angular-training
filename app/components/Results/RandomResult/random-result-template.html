<div ng-if="$ctrl.joke !== null">
    <reload-button-component on-reload="$ctrl.onReload" content="Get me another one !" visibility="$ctrl.joke !== null"></reload-button-component>

    <joke-component joke="$ctrl.joke"></joke-component>
</div>