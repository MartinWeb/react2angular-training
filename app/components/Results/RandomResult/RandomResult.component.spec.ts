import { RandomResultComponent } from "./RandomResult.component";

describe('RandomResultComponent', () => {
    test('should have valid snapshot', () => {
        expect(RandomResultComponent.template).toMatchSnapshot();
    });

    test('should have valid bindings', () => {
        expect(RandomResultComponent.bindings).toMatchSnapshot();
    });
});