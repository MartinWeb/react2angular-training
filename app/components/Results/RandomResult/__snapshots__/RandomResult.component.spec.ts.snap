// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomResultComponent should have valid bindings 1`] = `
{
  "joke": "=",
  "onReload": "<",
}
`;

exports[`RandomResultComponent should have valid snapshot 1`] = `
"<div ng-if="$ctrl.joke !== null">
    <reload-button-component on-reload="$ctrl.onReload" content="Get me another one !" visibility="$ctrl.joke !== null"></reload-button-component>

    <joke-component joke="$ctrl.joke"></joke-component>
</div>"
`;
